### ISSUE TITLE
SSL not working, lack of documentation

### ISSUE THREAD

**Issue by wdegeus** [gearman/gearmand#207]
First bug: starting gearmand  in SSL mode when it hasn't been compiled with ssl support works without throwing issues. (and subsequently it accepts plain text connections just fine)

Next, I finally have gearmand running with a self-signed CA and certificate, can connect to it just fine using openssl s_client.
When I try to connect using gearadmin or the gearman client however it dies with an undebuggable
```Error: error:00000001:lib(0):func(0):reason(1)```
The certificates are fine but self-signed, but I have no idea what its desires are.
How can I get this to work?
Is there an option to skip Common Name (CN) verification, if it does that at all?
Basically: there is almost no documentation on this at all. This page http://gearman.info/gearmand/ssl.html is almost useless.

**Comment by p-alik** [comment]
> So I strongly agree the documentation should be improved. Unfortunately, nobody ever answered that question for me, so I still don't know how to update
http://gearman.info/gearmand/ssl.html.

Moving the documentation to [github-wiki](https://github.com/gearman/gearmand/wiki) wouldn't be an option?

**Comment by esabol** [comment]
> > So I strongly agree the documentation should be improved. Unfortunately, nobody ever answered that question for me, so I still don't know how to update
> > http://gearman.info/gearmand/ssl.html.
> 
> Moving the documentation to [github-wiki](https://github.com/gearman/gearmand/wiki) wouldn't be an option?

I don't know. Have you looked at https://github.com/gearman/gearmand/pull/144 ?

**Comment by p-alik** [comment]
Github Pages is a gut alternative. If PR will be merged, the documentation will be available at https://gearman.github.io/gearmand/docs/dev/html/. But someone who has access to settings of gearman organisation should generate appropriate token and configure [GITHUB_TOKEN](https://github.com/BasBastian/gearmand/blob/master/.travis.yml#L61) environment variable in Travis CI.

https://docs.travis-ci.com/user/deployment/pages/
https://docs.travis-ci.com/user/environment-variables/#defining-variables-in-repository-settings

**Comment by esabol** [comment]
FYI, the source file for that document is here:
https://github.com/gearman/gearmand/blob/master/docs/source/gearmand/ssl.rst

@wdegeus, do have you any suggestions for what needs to be added to that documentation?

**Comment by esabol** [comment]
The second answer at https://stackoverflow.com/questions/10175812/how-to-create-a-self-signed-certificate-with-openssl is quite thorough.

**Comment by SpamapS** [comment]
I don't actually know who owns gearman.org or gearman.info .. @BrianAker or @dormando maybe? Anyway, we can pretty easily get those docs to build in to github pages and be available on gearman.org. However, I just started a new job and I'm pretty much on fumes for OSS contribution for the next few months... so somebody else will have to drive. :-P

**Comment by p-alik** [comment]
@SpamapS, could you generate an auth token add GITHUB_TOKEN in to Travis CI configuration, please.
See this [comment](#issuecomment-441376270) above.

**Comment by BrianAker** [comment]
While I provide support for the 2x branch, limited/personally, I am not sure what is need for v1.

**Comment by p-alik** [comment]
@BrianAker, may I ask you to take a look athttps://github.com/gearman/gearmand/issues/192#issuecomment-441114850, please. Memcache test fails sporadically in Travis CI. I'm just asking your opinion how it should be solved.

**Comment by illuusio** [comment]
Setting up SSL trust chain can be very harsh. Should we first have test case which creates foo certs and keys and then tests everything works?

**Comment by p-alik** [comment]
> Setting up SSL trust chain can be very harsh. Should we first have test case which creates foo certs and keys and then tests everything works?

There is no way around it, if ssl tests should be setup.

**Comment by illuusio** [comment]
Or do we have some certs which we use in git?

**Comment by p-alik** [comment]
Not yet but we could easily add some on demand.

**Comment by p-alik** [comment]
the issue was mentioned in https://stackoverflow.com/a/59072270/2789312

**Comment by esabol** [comment]
@p-alik : Are you able to rename issue titles? If so, please change the title of this issue from "SSL not working, lack of documentation" to "SSL documentation needs improvement" or similar. Thanks!
